dace/graph/nodes.py

```diff
@@ -921,3 +937,134 @@ def __label__(self, sdfg, state):
+            ret = cls(json_obj['attributes']['name'])
+            dace.serialize.set_properties_from_json(
+                ret, json_obj, context=context)
+            return ret
+
+    def expand(self, sdfg, *args, **kwargs):
+        """Create and perform the expansion transformation for this library
+           node."""
+        implementation = self.implementation
+        # If not explicitly set, try the node default
+        if implementation is None:
+            implementation = type(self).default_implementation
+        # If no node default, try library default
+        if implementation is None:
+            import dace.library  # Avoid cyclic dependency
+            lib = dace.library._DACE_REGISTERED_LIBRARIES[type(self)
+                                                          ._dace_library_name]
+            implementation = lib.default_implementation
+        # Otherwise we don't know how to expand
+        if implementation is None:
+            raise ValueError("No implementation or default "
+                             "implementation specified.")
+        if implementation not in self.implementations.keys():
+            raise KeyError("Unknown implementation: " + implementation)
+        transformation_type = type(self).implementations[implementation]
+        states = sdfg.states_for_node(self)
+        if len(states) < 1:
+            raise ValueError("Node \"" + str(self) +
+                             "\" not found in SDFG \"" + str(sdfg) + "\".")
+        if len(states) > 1:
+            raise ValueError("Node \"" + str(self) +
+                             "\" found in multiple states: " + ", ".join(
+                                 str(s) for s in states))
+        state = states[0]
+        sdfg_id = sdfg.sdfg_list.index(sdfg)
+        state_id = sdfg.nodes().index(state)
+        subgraph = {transformation_type._match_node: state.node_id(self)}
+        transformation = transformation_type(sdfg_id, state_id, subgraph, 0)
+        transformation.apply(sdfg, *args, **kwargs)
+
+    @classmethod
+    def register_implementation(clc, name, transformation_type):
+        """Register an implementation to belong to this library node type."""
+        clc.implementations[name] = transformation_type
+        match_node_name = "__" + transformation_type.__name__
+        if (hasattr(transformation_type, "_match_node")
+                and transformation_type._match_node != match_node_name):
+            raise ValueError(
+                "Transformation " + transformation_type.__name__ +
+                " is already registered with a different library node.")
+        transformation_type._match_node = clc(match_node_name)
+
+    def draw_node(self, sdfg, state):
+        return dot.draw_node(sdfg, state, self, shape="folder")
+
+    def validate(self, sdfg, state):
+        raise ValueError("Unexpanded library node of type " + str(type(self)) +
+                         ".")
+
+
+def static_connectors(clc):
```

Comment:
This is unused. Can it be removed? @alexnick83 @definelicht

Reply:
Sure, I like the idea, but it didn't work with our current setup. I removed it for TODO(later (maybe)).